library/alloc/src/raw_vec.rs

```diff
@@ -462,85 +657,26 @@ impl<T, A: Allocator> RawVec<T, A> {
     // of the code that doesn't depend on `T` as possible is in functions that
     // are non-generic over `T`.
     fn grow_amortized(&mut self, len: usize, additional: usize) -> Result<(), TryReserveError> {
+        const { assert!(mem::size_of::<T>() % mem::align_of::<T>() == 0) };
```

Comment:
🤷 This was in the original code. There's also a wrong comment in `current_memory` about types where size != stride, such types do not exist in Rust.

Reply:
#83706 added this check by going through `Layout::array`
#107167 in partially undid it since it previously caused a small perf regression but left the const assert in its place to move the check to compile time

Reply:
I've deleted all these const assertions. I don't know if that was the right move here? I'm not opposed to adding it by some other means, but it seems a bit silly to have an assertion if it's true by definition.